src/lib/dfp/init-slot-ias.ts

```diff
@@ -0,0 +1,128 @@
+import { once } from 'lodash-es';
+import { getUrlVars } from 'lib/utils/url';
+import type { IasPETSlot, IasTargeting } from 'types/ias';
+
+const adUnit = once((): string => {
+	const urlVars = getUrlVars();
+	return urlVars['ad-unit']
+		? `/${window.guardian.config.page.dfpAccountId}/${String(
+				urlVars['ad-unit'],
+		  )}`
+		: window.guardian.config.page.adUnit;
+});
+
+const timeout = async <T>(
+	promise: Promise<T>,
+	ms: number,
+): Promise<T | void> => {
+	let timeoutId: number;
+	return Promise.race([
+		promise,
+		new Promise<void>((resolve) => {
+			timeoutId = window.setTimeout(resolve, ms);
+		}),
+	]).then((result) => {
+		window.clearTimeout(timeoutId);
+		return result;
+	});
+};
+
+const initSlotIas = (id: string, slot: googletag.Slot) =>
+	timeout(
+		new Promise<void>((resolve) => {
+			/*
```

Comment:
The formatting of the multi line comments in this file looks a tad off, does Prettier not mind?